enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+  // zone. This ensures instances each have their own networking and power source
+  // for maximum hardware fault tolerance.
+  // Spread placement groups are recommended for a small number of critical instances
+  // which must be kept separate from one another.
+  // Using a Spread placement group imposes a limit of seven instances within
+  // the placement group within a single availability zone.
+  AWSSpreadPlacementGroupType    AWSPlacementGroupType = "Spread"
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Placement Group Strategies
+
+AWS offers three different placement group strategies.
+Each strategy is designed with a different use case in mind and each has it's own caveats.
+The details of each are fleshed out below.
+
+##### Cluster Placement Groups
+
+A Cluster placement group is a grouping of instances within a single Availability zone.
+Cluster placement groups are used for workloads that require higher per-flow throughput limits for TCP/IP traffic and
+as such, are placed in the same high-bisection bandwidth segment of the network.
+
+Cluster placement groups are recommended for applications that benefit from low network latency, high network
+throughput, or both.
+In particular, we expect that these will be used in conjunction with EFA network interfaces in HPC applications.
+
+When using Cluster placement groups, AWS recommends to launch all instances as the same instance type and if
+possible, to launch instances simultaneously to prevent the likelihood that capacity limits are reached within the
+placement group.
+
+###### Limitations of Cluster Placement Groups
+
+- The following instance types are supported:
+  - Current generation instances, except for burstable performance instances (for example, T2) and Mac1 instances.
+  - The following previous generation instances: `A1`, `C3`, `cc2.8xlarge`, `cr1.8xlarge`, `G2`, `hs1.8xlarge`, `I2`,
+    and `R3`.
+- A cluster placement group can't span multiple Availability Zones.
+- You can launch multiple instance types into a cluster placement group. However, this reduces the likelihood that the
+  required capacity will be available for your launch to succeed. We recommend using the same instance type for all
+  instances in a cluster placement group.
+
+##### Spread Placement Groups
+
+A Spread placement group is a grouping of instances all placed in distinct racks within the datacenter.
+This ensures that each instance within the group has its own network and power source, maximising resillience to
+hardware failures.
+
+Spread placement groups are recommended for launching a small number of instances that must be kept separate, eg a
+cluster of etcd instances.
+
+As Spread placement groups keep instances in different racks, they can be expanded over time and can also mix instance
+types without detriment.
+
+Spread placement groups have a limited capacity. You can only launch seven instances within a Spread placement group
+within a single availability zone.
```

Comment:
can I use it for muitiple availability zones normally?

Reply:
There's nothing to stop you using it across multiple availability zones, but a MachineSet is tied to a single AZ, so each MachineSet gains a limit of seven instances when it has a spread placement group (unless they create more than one machineset in the same zone, in which case, fewer than 7 I would expect)